heroic-component/src/main/java/com/spotify/heroic/metric/QueryTrace.java

```diff
@@ -22,128 +22,517 @@
+     */
+    static Identifier identifier(Class<?> where) {
+        return new Identifier(where.getName());
     }
 
-    public void formatTrace(String prefix, PrintWriter out) {
-        out.println(prefix + what + " (in " + readableTime(elapsed) + ")");
-
-        for (final QueryTrace child : children) {
-            child.formatTrace(prefix + "  ", out);
-        }
+    /**
+     * Create a new identifier for a class and a what
+     *
+     * @param where Class associated with the identifier
+     * @param what String describing what is being traced
+     * @return an {@link com.spotify.heroic.metric.QueryTrace.Identifier}
+     */
+    static Identifier identifier(Class<?> where, String what) {
+        return new Identifier(where.getName() + "#" + what);
     }
 
-    public static Identifier identifier(Class<?> where) {
-        return new Identifier(where.getName());
+    /**
+     * Create a new identifier for a free text description.
+     *
+     * @param description Description telling what is being traced.
+     * @return an {@link com.spotify.heroic.metric.QueryTrace.Identifier}
+     */
+    static Identifier identifier(String description) {
+        return new Identifier(description);
     }
 
-    public static Identifier identifier(Class<?> where, String what) {
-        return new Identifier(where.getName() + "#" + what);
+    /**
+     * Create a new watch.
+     *
+     * @return a {@link com.spotify.heroic.metric.QueryTrace.Watch}
+     * @deprecated use {@link Tracing#watch()}
+     */
+    static Watch watch() {
+        return new ActiveWatch(Stopwatch.createStarted());
     }
 
-    public static Identifier identifier(String description) {
-        return new Identifier(description);
+    /**
+     * Create a new watch.
+     *
+     * @return a {@link com.spotify.heroic.metric.QueryTrace.Watch}
+     * @deprecated use {@link Tracing#watch(Identifier)}
+     */
+    static NamedWatch watch(final Identifier what) {
+        return new ActiveNamedWatch(what, Stopwatch.createStarted());
     }
 
-    private String readableTime(long elapsed) {
-        if (elapsed > 1000000000) {
-            return String.format("%.3fs", elapsed / 1000000000d);
+    /**
+     * Format the current trace onto the given PrintWriter.
+     *
+     * @param out {@link java.io.PrintWriter} to format to
+     */
+    void formatTrace(PrintWriter out);
+
+    /**
+     * Format the current trace onto the given PrintWriter with the given prefix.
+     *
+     * @param out {@link java.io.PrintWriter} to format to
+     * @param prefix prefix to prepend
+     */
+    void formatTrace(PrintWriter out, String prefix);
+
+    /**
+     * @deprecated use {@link #elapsed()}
+     */
+    @Deprecated
+    long getElapsed();
+
+    /**
+     * @deprecated use {@link #what()}
+     */
+    @Deprecated
+    Identifier getWhat();
+
+    /**
+     * @deprecated use {@link #children()}
+     */
+    @Deprecated
```

Comment:
What is the reasoning behind deprecating these methods and replacing them? Shorter method name?